generic/tests/monotonic_time.py

```diff
@@ -0,0 +1,84 @@
+import os
+import logging
+
+from virttest import data_dir
+from virttest import error_context
+
+
+class TimeClientTest(object):
+    def __init__(self, test, params, env, test_name):
+        self.test = test
+        self.vm_params = params
+        self.env = env
+        self.vm = self.env.get_vm(self.vm_params["main_vm"])
+        self.session = self.vm.wait_for_login()
+        self.host_dir = os.path.join(data_dir.get_deps_dir(), test_name)
+        self.src_dir = os.path.join("/tmp/", test_name)
+
+    def setUp(self):
+        logging.info("Copy files to guest")
+        self.vm.copy_files_to(self.host_dir, "/tmp/")
+        self.session.cmd("cd %s; make clobber; make;" % self.src_dir)
+
+    def cleanUp(self):
+        self.session.cmd("rm -rf %s" % self.src_dir)
+        self.session.close()
+
+
+class MonotonicTime(TimeClientTest):
+    def __init__(self, test, params, env, test_name):
+        super(MonotonicTime, self).__init__(test, params, env, test_name)
+
+    def setUp(self):
+        super(MonotonicTime, self).setUp()
+
+    def _test(self, test_type=None, duration=300, threshold=None):
+        if not test_type:
+            self.test.error('missing test type')
+        logging.info("Test type: %s" % test_type)
+        timeout = float(duration) + 100.0
+
+        cmd = self.src_dir + '/time_test'
+        cmd += ' --duration ' + str(duration)
+        if threshold:
+            cmd += ' --threshold ' + str(threshold)
+        cmd += ' ' + test_type
+
+        (self.exit_status, self.stdout) = self.session.cmd_status_output(cmd, timeout=timeout)
```

Comment:
If `exit_status` and `stdout` are treated as the member of the `Test` class, please initialize them in `__init__`, otherwise,  not need to assign them to `self`.